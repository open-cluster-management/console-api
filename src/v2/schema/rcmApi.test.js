/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('RCM Api Resolver', () => {
  test('Get Automated Import Status', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          {
            getAutomatedImportStatus(namespace: "test", name:"test")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Get Automated Import Status should return error msg', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          {
            getAutomatedImportStatus(namespace: "", name:"test")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Create Kubernetes Cluster Resource for Import', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
            createClusterResource(body: "test")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Create Kubernetes Cluster Resource for Import should return error msg', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
            createClusterResource(body: "")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Start Automated Import Process', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
            automatedImport(
              namespace:"test",
              name:"test",
              body: "test"
              )
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Start Automated Import Process should return error msg', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
            automatedImport(namespace:"", name:"test", body: "test")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Correctly Resolves Get Cloud Providers', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            {
              orchestrations {
                name
                longname
                type
                configMetadata
                configValues
                clusterMetadata
                clusterValues
                categoryName
                categoryLongname              
                statusCode
              }
            }`,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('createCluster test that we can create cluster correctly', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                createCluster(namespace: "iks", cluster: {name: "ericabrtest4"})
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
  test('createCluster test with all required fields but returns with error', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                createCluster(namespace: "iks", cluster: {name:"ericabrtest5"})
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('createCluster test that we get an error when namespace not provided', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                createCluster(namespace:"", {name:"ericabrtest4"})
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('createCluster test that we get an error when cluster not provided', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                createCluster(namespace:"", cluster: null)
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('previewCluster test with all required fields but returns with error', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                previewCluster(namespace: "eks", cluster: {name:"llcao-aks-1"})
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('previewCluster test that we get an error when namespace not provided', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                previewCluster(namespace:"", {name:"llcao-gke-1"})
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('previewCluster test that we get an error when cluster not provided', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
            mutation {
                previewCluster(namespace:"", cluster: null)
            }
          `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Update Kubernetes Cluster Resource for Import', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
          updateClusterResource(namespace:"test", name:"test", body: "test")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Update Kubernetes Cluster Resource for Import should return error msg', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
          updateClusterResource(namespace:"", name:"test", body: "test")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
});
