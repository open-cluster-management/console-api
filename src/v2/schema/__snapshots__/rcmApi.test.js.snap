// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RCM Api Resolver Create Kubernetes Cluster Resource for Import 1`] = `
Object {
  "data": Object {
    "createClusterResource": Object {
      "kind": "Secret",
    },
  },
}
`;

exports[`RCM Api Resolver Create Kubernetes Cluster Resource for Import should return error msg 1`] = `
Object {
  "data": Object {
    "createClusterResource": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 13,
          "line": 3,
        },
      ],
      "message": "Body is required for createClusterResource",
      "path": Array [
        "createClusterResource",
      ],
    },
  ],
}
`;

exports[`RCM Api Resolver Update Kubernetes Cluster Resource for Import 1`] = `
Object {
  "data": Object {
    "updateClusterResource": Object {
      "apiVersion": "clusterregistry.k8s.io/v1alpha1",
      "items": Array [
        Object {
          "apiVersion": "clusterregistry.k8s.io/v1alpha1",
          "kind": "Cluster",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2018-08-13T16:56:41Z",
            "finalizers": Array [
              "finalizer.hcm.ibm.com",
            ],
            "labels": Object {
              "cloud": "IBM",
              "datacenter": "toronto",
              "environment": "Dev",
              "name": "managed-cluster",
              "owner": "marketing",
              "region": "US",
              "vendor": "ICP",
            },
            "name": "managed-cluster",
            "namespace": "default",
            "resourceVersion": "136558",
            "selfLink": "/apis/clusterregistry.k8s.io/v1alpha1/namespaces/default/clusters/mycluster.icp",
            "uid": "d9f3a5aa-9f19-11e8-855e-f2b998610544",
          },
          "spec": Object {
            "authInfo": Object {},
            "kubernetesApiEndpoints": Object {
              "serverEndpoints": Array [
                Object {
                  "serverAddress": "2.2.2.2:8001",
                },
              ],
            },
          },
          "status": Object {},
        },
        Object {
          "apiVersion": "clusterregistry.k8s.io/v1alpha1",
          "kind": "Cluster",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2018-08-13T16:56:41Z",
            "finalizers": Array [
              "finalizer.hcm.ibm.com",
            ],
            "labels": Object {
              "cloud": "IBM",
              "datacenter": "raleigh",
              "environment": "Dev",
              "name": "hub-cluster",
              "owner": "development",
              "region": "US",
              "vendor": "ICP",
            },
            "name": "hub-cluster",
            "namespace": "kube-system",
            "resourceVersion": "136558",
            "selfLink": "/apis/clusterregistry.k8s.io/v1alpha1/namespaces/kube-system/clusters/hub-cluster",
            "uid": "d9f3a5aa-9f19-11e8-855e-f2b998610544",
          },
          "spec": Object {
            "authInfo": Object {},
            "kubernetesApiEndpoints": Object {
              "serverEndpoints": Array [
                Object {
                  "serverAddress": "1.1.1.1:8001",
                },
              ],
            },
          },
          "status": Object {
            "conditions": Array [
              Object {
                "lastHeartbeatTime": "2018-08-15T19:41:20Z",
                "lastTransitionTime": null,
                "status": "",
                "type": "OK",
              },
            ],
          },
        },
        Object {
          "apiVersion": "clusterregistry.k8s.io/v1alpha1",
          "kind": "Cluster",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2018-08-13T16:56:41Z",
            "finalizers": Array [
              "finalizer.hcm.ibm.com",
            ],
            "labels": Object {
              "cloud": "IBM",
              "datacenter": "raleigh",
              "environment": "Dev",
              "name": "new-cluster",
              "owner": "development",
              "region": "US",
              "vendor": "ICP",
            },
            "name": "new-cluster",
            "namespace": "kube-system",
            "resourceVersion": "137558",
            "selfLink": "/apis/clusterregistry.k8s.io/v1alpha1/namespaces/kube-system/clusters/new-cluster",
            "uid": "d9f3a6aa-9f19-11f8-855e-f2b908610543",
          },
          "spec": Object {
            "authInfo": Object {},
            "kubernetesApiEndpoints": Object {
              "serverEndpoints": Array [
                Object {
                  "serverAddress": "3.3.3.3:8001",
                },
              ],
            },
          },
          "status": Object {
            "conditions": Array [
              Object {
                "lastHeartbeatTime": "2018-08-15T19:41:20Z",
                "lastTransitionTime": null,
                "status": "",
                "type": "",
              },
            ],
          },
        },
      ],
      "kind": "ClusterList",
      "metadata": Object {
        "resourceVersion": "136667",
        "selfLink": "/apis/clusterregistry.k8s.io/v1alpha1/clusters",
      },
      "statusCode": "",
      "statusMessage": "",
    },
  },
}
`;

exports[`RCM Api Resolver Update Kubernetes Cluster Resource for Import should return error msg 1`] = `
Object {
  "data": Object {
    "updateClusterResource": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Body, namespace, and name are required for updateClusterResource",
      "path": Array [
        "updateClusterResource",
      ],
    },
  ],
}
`;
